xamarin/Xamarin.Forms
# [Android] Fix brief display of incorrect keyboard

### Description of Change ###

Proposing this as an alternative to #4248 and a solution to #3647.

Rather than making `HandleKeyboardOnFocus` available to custom renderers, this change makes the `OnFocusChangeRequested` handler available and implements overrides in the appropriate classes for showing/hiding the keyboard (rather than handling it in `ViewRenderer`). 

This change also posts requests to show the keyboard to the main looper queue, thus matching them up with the posted requests for control focus; this fixes the "wrong keyboard briefly displayed" bug. 

Previously, the focus requests were delayed but the keyboard display happened within `OnFocusChangeRequested`; this meant that `ShowKeyboard` was called when the target control was not actually focused, and `ShowSoftInput` always failed, requiring a call to `ToggleSoftInput` to force the keyboard open. The `ToggleSoftInput` call caused the last or default keyboard to be displayed briefly until the main looper processed the focus request and displayed the correct keyboard for the input.

By posting the "show keyboard" requests to the queue after the "focus" requests, they happen in the correct order and `ShowKeyboard` is called with the correct control having focus; `ShowSoftInput` then runs successfully.

### Issues Resolved ### 

- fixes #3647 

### API Changes ###

Android, ViewRenderer.cs:

`internal virtual void OnFocusChangeRequested(object sender, VisualElement.FocusRequestArgs e)` ->
`protected virtual void OnFocusChangeRequested(object sender, VisualElement.FocusRequestArgs e)`

### Platforms Affected ### 

- Android

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###

Programmatically switch focus between an Entry/Editor/SearchBar with a Numeric keyboard and an Entry/Editor/SearchBar with an Email keyboard several dozen times; verify that the incorrect keyboard does not flash briefly onscreen.

### PR Checklist ###
